src/test/rgw/test_rgw_cksum.cc

```diff
@@ -0,0 +1,329 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include <errno.h>
+#include <iostream>
+#include <fstream>
+#include <string>
+
+#include "gtest/gtest.h"
+
+#include "common/config.h"
+#include "common/ceph_argparse.h"
+#include "common/debug.h"
+#include "rgw/rgw_cksum.h"
+#include <openssl/sha.h>
+#include "rgw/rgw_hex.h"
+
+#define dout_subsys ceph_subsys_rgw
+
+namespace {
+
+  using namespace rgw;
+  using namespace rgw::cksum;
+
+  bool verbose = false;
+
+  cksum::Type t1 = cksum::Type::blake3;
+  cksum::Type t2 = cksum::Type::sha1;
+  cksum::Type t3 = cksum::Type::sha256;
+  cksum::Type t4 = cksum::Type::sha512;
+  cksum::Type t5 = cksum::Type::crc32;
+  cksum::Type t6 = cksum::Type::crc32c;
+  cksum::Type t7 = cksum::Type::xxh3;
+
+  std::string lorem =
+    "Lorem ipsum dolor sit amet";
+
+  std::string dolor =
+    R"(Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.)";
+
+TEST(RGWCksum, Output)
+{
+  auto o_mode = std::ios::out|std::ios::trunc;
+  std::ofstream of;
+  of.open("/tmp/lorem", o_mode);
+  of << lorem;
+  of.close();
+
+  of.open("/tmp/dolor", o_mode);
+  of << dolor;
+  of.close();
+}
+
+TEST(RGWCksum, DigestCRC32)
+{
+  auto t = cksum::Type::crc32;
+  DigestVariant dv = rgw::cksum::digest_factory(t);
+  Digest* digest = get_digest(dv);
+
+  ASSERT_NE(digest, nullptr);
+
+  digest->Update((const unsigned char *)dolor.c_str(), dolor.length());
+
+  auto cksum = rgw::cksum::finalize_digest(digest, t);
+  
+  /* compare w/known value https://crccalc.com/ */
+  ASSERT_EQ(cksum.hex(), "98b2c5bd" /* not quite bdc5b298 */);
+  /* compare w/known value https://www.base64encode.org/ */
+  ASSERT_EQ(cksum.to_base64(), "OThiMmM1YmQ=");
+}
+
+TEST(RGWCksum, DigestCRC32c)
+{
+  auto t = cksum::Type::crc32c;
+  DigestVariant dv = rgw::cksum::digest_factory(t);
+  Digest* digest = get_digest(dv);
+
+  ASSERT_NE(digest, nullptr);
+
+  digest->Update((const unsigned char *)dolor.c_str(), dolor.length());
+
+  auto cksum = rgw::cksum::finalize_digest(digest, t);
+  /* compare w/known value https://crccalc.com/ */
+  ASSERT_EQ(cksum.hex(), "4b2edc95");
```

Comment:
following previous comment regarding `byteswap()` [1]
suggesting a change to reverse the byteorder: 
`ASSERT_EQ(cksum.hex(), "95DC2E4B");`
to be consistent with the display in:
https://crccalc.com/?crc=Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.&method=crc32&datatype=ascii&outtype=0
![image](https://github.com/ceph/ceph/assets/31659604/6e0e0d38-efa2-4dcd-9eee-8f64429003e5)


[1] https://github.com/ceph/ceph/pull/54584/files#r1402416023

Reply:
oh, I guess I reversed it myself!

Reply:
I thought the current code was doing that already without byteswap

Reply:
will recheck and fix

Reply:
my intent was to match that value, which you also produced with crc32 and cksfv